src/server/resc_attr.c

```diff
@@ -923,35 +935,47 @@ action_soft_walltime(resource *presc, attribute *pattr, void *pobject, int type,
 		pjob = (job *) pobject;
 
 		/* Make sure soft_walltime < walltime */
-		def = find_resc_def(svr_resc_def, WALLTIME, svr_resc_size);
-		if (!def)
+		if (soft_walltime_def == NULL)
+			soft_walltime_def = find_resc_def(svr_resc_def, WALLTIME, svr_resc_size);
+		if (soft_walltime_def == NULL)
```

Comment:
This is an unlikely case but I think we should return something like PBSE_NORESC. Unfortunately that error message is not there, so the closest thing to it is PBSE_NOATTR.

Reply:
I followed what was being done for STF jobs here.  I have an alternate solution.  How about if the resource def can't be found, we just skip that check (not return success).  I think this is the right thing to do.  The only way to get a NULL is if someone were to remove a resource.  Since these checks enforce a dependency between two resources, that dependency can't fail if one resource doesn't exist.